### ISSUE TITLE
BRSL: deriveAll failing for equation containing noEvent and floor

### ISSUE THREAD

**Issue by AnHeuermann** [OpenModelica/OpenModelica#9690]
### Description

Found an issue while investigating #9131. When enabeling symbolic Jaconian for the test model `Rexroth_Valves_Test.Valves_4WR.Rexroth_4WSE2ED10_5X_5.VC43_PO_TC11a` the symbolica Jacobian can't be created.

### Steps to Reproduce

MWE using BRSL v2.2dev (from 09-09-2022):
```modelica
loadFile("path/to/BRSL/Rexroth_Generic/package.mo"); getErrorString();
loadFile("path/to/BRSL/Rexroth_Valves/package.mo"); getErrorString();
loadFile("path/to/Rexroth_Valves_Test/package.mo"); getErrorString();

setCommandLineOptions("--generateSymbolicJacobian"); getErrorString();
buildModel(Rexroth_Valves_Test.Valves_4WR.Rexroth_4WSE2ED10_5X_5.VC43_PO_TC11a); getErrorString();
```

```
[/path/to/OpenModelica/OMCompiler/Compiler/BackEnd/Differentiate.mo:267:5-267:157:writable] Error: Derivative of expression \"$cse12 = noEvent(floor(0.4 * testSignal.clock.y))\" w.r.t. \"dummyVarA\" is non-existent.
Error: Internal error SymbolicJacobian.deriveAll failed
[/path/to/OpenModelica/OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo:2440:7-2440:87:writable] Error: Internal error function generateSymbolicJacobian failed
/path/to/OpenModelica/OMCompiler/Compiler/BackEnd/SymbolicJacobian.mo:2231:9-2231:79:writable] Error: Internal error function createJacobian failed
Error: post-optimization module symbolicJacobian (simulation) failed.
```

### Expected Behavior

I'm not sure. What the derivative should be in this case. `noEvent` can probably be ignored, but what's with rounding function `floor`?
But in the end it is probably just `0`?

### Version and OS

 - OpenModelica Version: v1.21.0-dev-11-g3047cded7c
 - OS: Ubuntu 20.04.5 LTS
 - Versions of used Modelica libraries:
   - BRSL v2.2dev (from 09-09-2022)

**Comment by AnHeuermann** [comment]
@kabdelhak and myself tried to figure out what's going on here.

We have a discrete equation
```
$cse12 = noEvent(floor(0.4 * testSignal.clock.y))
testSignal.firstOrder.u = 1.0 + (0.01 + 0.01 * $cse12 + testSignal.y1 + testSignal.y2) * testSignal.Dp_flow_load
```

in the right hand side of the DAE.
If $cse12 jumps at simulation time the derivative is undefined and naturally an ODE sovler should have issue when trying to compute the derivative to close to the discrete jump.
I'm not sure what should happen in this case.

**Comment by casella** [comment]
There are several potentially intertwined issues here.

First of all, `floor(0.4*testSignal.clock.y)` is a discontinuous function. Placing `noEvent()` around it suppresses event detection and handling, so it means that the ODE solver will be passed a RHS which is discontinuous. If the the ODE solver uses high-order integration algorithms and error estimations, those won't work correctly, because they assume that the solution has a certain smoothness. In some cases a variable step-size solver would detect that the error is very high, and interpret it as the result of an extremely non-linear RHS, so it will reduce the step, producing a somewhat accurate result. In other cases, it may simply skip the event. There is a very nice description of this issue in Francois Cellier's Continuous System Simulation book, Chapter 9, please have a look at that.

Francois' conclusion is that in most cases it is eventually better to use proper event detection, because then the solver works on functions whose actual mathematical properties are consistent with the assumptions of the solver algorithms. However, in some cases one may want to skip the events for some reason, since Modelica allows that, we should try support it.

So, assuming the RHS is discontinuous because of noEvent, the next question is, can we compute a symbolic Jacobian for a stiff solver? In principle the answer is yes, around the discontinuity you will pass from a left-limit derivative to a right-limit derivative. All the theory on which implicit solvers are based upon will be shaking a bit when the simulation passes through these singular points, but as long as the users is OK with that, so be it.

BTW, we are already doing something worse with numerical Jacobians, because in that case you don't get the left and right limit, but rather an approximation of an impulse, because of the discontinuity of the function. So, allowing symbolic differentiation could actually improve the situation.

All that being said, in this specific case the RHS is discontinuous, _but only over time_, not with respect to the state. So I actually see no reason to fail the symbolic differentiation _a fortiori_

Bottom line, using noEvent on discontinuous functions is questionable from a solver point of view, but if the library author is fine with that, just go ahead, and consider the derivative of floor() to be zero, _when generating the ODE RHS jacobian_

The issue is completely different when you are differentiating for index reduction. In that case, differentiating a constraint function that is discontinuous with respect to the states would lead to a plain wrong solution, and should generate an error, see e.g. [this comment](https://github.com/OpenModelica/OpenModelica/issues/8658#issue-1163023040).
